cedar-lean/Cedar/Partial/Evaluator.lean

```diff
@@ -0,0 +1,159 @@
+/-
+ Copyright 2022-2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-/
+
+import Cedar.Partial.Entities
+import Cedar.Partial.Expr
+import Cedar.Partial.Request
+import Cedar.Partial.Value
+import Cedar.Spec.Evaluator
+import Cedar.Spec.ExtFun
+import Cedar.Spec.Value
+
+/-! This file defines the semantics of Cedar partial evaluation. -/
```

Comment:
Would be useful to have comments here about which functions are analogous to the ones in `Spec.Evaluator`, and which ones are different.

Reply:
Will do in a followup

Reply:
#286